src/openforms/forms/utils.py

```diff
@@ -319,12 +323,19 @@ def parse_trigger(
         trigger_component_key = trigger.get("when")
         compare_value = trigger.get("eq")
 
-        if (
-            configuration_wrapper
-            and configuration_wrapper[trigger_component_key]["type"] == "selectboxes"
-        ):
-            return {"==": [{"var": f"{trigger_component_key}.{compare_value}"}, True]}
-
+        if configuration_wrapper:
+            if trigger_component_key not in configuration_wrapper:
+                logger.warning(
+                    "Logic trigger with non-existing component (%s)",
+                    trigger_component_key,
+                )
+                return
+            else:
+                trigger_component_type = configuration_wrapper[trigger_component_key][
+                    "type"
+                ]
+                if trigger_component_type == "selectboxes":
+                    trigger_component_key = f"{trigger_component_key}.{compare_value}"
```

Comment:
```suggestion
                if trigger_component_type == "selectboxes":
                    trigger_component_key = f"{trigger_component_key}.{compare_value}"
                    compare_value = True
```
this bit is missing to pass the existing test